kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -223,6 +233,44 @@ def _create_kubelet_prometheus_instance(self, instance):
         )
         return kubelet_instance
 
+    def _create_pod_tags_by_pvc(self, pods):
+        """
+        Return a map, e.g.
+            {
+                "<persistentvolumeclaim>|<kube_namespace>": [<list_of_pod_tags>],
+                "<persistentvolumeclaim1>|<kube_namespace1>": [<list_of_pod_tags1>],
+            }
+        that can be used to add pod tags to associated volume metrics
+        """
+        pod_tags_by_pvc = {}
+        for pod in pods['items']:
+            # get persistentvolumeclaim
+            volumes = pod.get('spec', {}).get('volumes')
+            if not volumes:
+                continue
+            for v in volumes:
+                pvc_name = v.get('persistentVolumeClaim', {}).get('claimName')
+                if pvc_name:
+                    break
```

Comment:
what if the pod is attached to more than one PVC? could you please adapt the logic to collect all PVCs and add some unit tests with pods having multiple PVCs attached? 

```
  volumes:
  - name: cassandra-data
    persistentVolumeClaim:
      claimName: cassandra-data-cassandra-0
  - name: another-cassandra-data
    persistentVolumeClaim:
      claimName: another-cassandra-data-cassandra-0
```